shared/actions/teams.js

```diff
@@ -133,86 +132,79 @@ const _leaveTeam = function(state: TypedState, action: TeamsGen.LeaveTeamPayload
-  }
+  return RPCTypes.teamsTeamAddMembersRpcPromise(
+    {
+      assertions: ids,
+      name: teamname,
+      role:
+        RPCTypes.teamsTeamRole[role] === undefined
+          ? RPCTypes.teamsTeamRole.none
+          : RPCTypes.teamsTeamRole[role],
+      sendChatNotification,
+    },
+    Constants.teamWaitingKey(teamname)
+  )
+    .then(() => {
+      // Success, dismiss the create team dialog and clear out search results
+      logger.info(`Successfully added ${ids.length} users to ${teamname}`)
+      return [
+        RouteTreeGen.createPutActionIfOnPath({
+          expectedPath: rootPath.concat(sourceSubPath),
+          otherAction: RouteTreeGen.createNavigateTo({parentPath: rootPath, path: destSubPath}),
+          parentPath: rootPath,
+        }),
+        SearchGen.createClearSearchResults({searchKey: 'addToTeamSearch'}),
+        SearchGen.createSetUserInputItems({searchKey: 'addToTeamSearch', searchResults: []}),
+        TeamsGen.createSetTeamInviteError({error: ''}),
+      ]
+    })
+    .catch(error => {
+      logger.error(`Error adding to ${teamname}: ${error.desc}`)
+      // Some errors, leave the search results so user can figure out what happened
+      logger.info(`Displaying addPeopleToTeam errors...`)
+      return TeamsGen.createSetTeamInviteError({error: error.desc})
+    })
 }
 
-const _getTeamRetentionPolicy = function*(action: TeamsGen.GetTeamRetentionPolicyPayload) {
+const getTeamRetentionPolicy = (state, action) => {
   const {teamname} = action.payload
-  const state = yield* Saga.selectState()
   const teamID = Constants.getTeamID(state, teamname)
   if (!teamID) {
     const errMsg = `getTeamRetentionPolicy: Unable to find teamID for teamname ${teamname}`
     logger.error(errMsg)
     return
   }
-  const policy = yield* Saga.callPromise(
-    RPCChatTypes.localGetTeamRetentionLocalRpcPromise,
-    {teamID},
-    Constants.teamWaitingKey(teamname)
-  )
+
   let retentionPolicy = Constants.makeRetentionPolicy()
-  try {
-    retentionPolicy = Constants.serviceRetentionPolicyToRetentionPolicy(policy)
-    if (retentionPolicy.type === 'inherit') {
-      throw new Error(`RPC returned retention policy of type 'inherit' for team policy`)
-    }
-  } catch (err) {
-    logger.error(err.message)
-    throw err
-  } finally {
-    yield Saga.put(TeamsGen.createSetTeamRetentionPolicy({retentionPolicy, teamname}))
-  }
+  return RPCChatTypes.localGetTeamRetentionLocalRpcPromise({teamID}, Constants.teamWaitingKey(teamname))
+    .then(policy => {
+      try {
+        retentionPolicy = Constants.serviceRetentionPolicyToRetentionPolicy(policy)
+        if (retentionPolicy.type === 'inherit') {
+          throw new Error(`RPC returned retention policy of type 'inherit' for team policy`)
+        }
+      } catch (err) {
+        logger.error(err.message)
+        throw err
+      }
+    })
+    .finally(() => TeamsGen.createSetTeamRetentionPolicy({retentionPolicy, teamname}))
```

Comment:
I think something is up with this, retention pickers for teams are perma-spinning in team settings + info panel

Reply:
yeah this is incorrect, you can't return from finally